test/tests/functional/pbs_test_tpp.py

```diff
@@ -1003,6 +1004,48 @@ def test_isolated_mom_pools_when_comm_on_non_serverhost(self):
         self.set_pbs_conf(host_name=self.server.shortname, conf_param=c)
         self.common_steps_for_mom_pool_tests()
 
+    @requirements(num_moms=2)
+    def test_comm_instantiation_on_cmdline(self):
+        """
+        Test pbs_comm instantiation through command line
+        Configuration:
+        Node 1 : Server, Sched, Mom, Comm
+        Node 2 : Mom
+        """
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+        self.comm.signal("-KILL")
+        for mom in self.moms.values():
+            self.server.expect(NODE, {'state': 'state-unknown,down'},
+                               id=mom.shortname)
+        self.logger.info("starting pbs_comm through command line")
+        pbs_comm_path = os.path.join(self.pbs_conf['PBS_EXEC'], "sbin",
+                                     "pbs_comm")
+        cmd = "sudo -u root %s -N" % pbs_comm_path
+        process = subprocess.Popen(cmd, shell=True)
```

Comment:
why do we need shell=True?
also it would be good if we do this in a try: except.